cryptofeed/backends/redis.py

```diff
@@ -32,38 +35,107 @@ def __init__(self, host='127.0.0.1', port=6379, socket=None, key=None, numeric_t
 
         self.redis = aioredis.from_url(f"{prefix}{host}:{port}")
         self.key = key if key else self.default_key
+        self.recentOnly = self.recentOnly
+        self.authorized = self.authorized
         self.numeric_type = numeric_type
 
-
-class RedisZSetCallback(RedisCallback):
+class RedisHashCallback(RedisCallback):
     def __init__(self, host='127.0.0.1', port=6379, socket=None, key=None, numeric_type=float, score_key='timestamp', **kwargs):
-        """
-        score_key: str
-            the value at this key will be used to store the data in the ZSet in redis. The
-            default is timestamp. If you wish to look up the data by a different value,
-            use this to change it. It must be a numeric value.
-        """
         self.score_key = score_key
         super().__init__(host=host, port=port, socket=socket, key=key, numeric_type=numeric_type, **kwargs)
-
+    
     async def write(self, feed: str, symbol: str, timestamp: float, receipt_timestamp: float, data: dict):
         score = data[self.score_key]
-        data = json.dumps(data)
         await self.queue.put({'feed': feed, 'symbol': symbol, 'score': score, 'data': data})
-
+    
     async def writer(self):
         while True:
-
             count = self.queue.qsize()
             if count > 1:
                 async with self.read_many_queue(count) as updates:
                     async with self.redis.pipeline(transaction=False) as pipe:
+                        print(f"count >1: RecentOnly-{self.recentOnly}; Authorized-{self.authorized}")
                         for update in updates:
-                            pipe = pipe.zadd(f"{self.key}-{update['feed']}-{update['symbol']}", {update['data']: update['score']}, nx=True)
+                            if self.authorized:
+                                for k, v in update['data'].items():
+                                    redis_v = pipe.hget(f"{self.key}-{update['data']['account']}-{update['feed']}-{update['symbol']}", k)
+                                    if redis_v is not None and redis_v != str(v):
+                                        pipe = pipe.hset(f"{self.key}-{update['data']['account']}-{update['feed']}-{update['symbol']}", k, str(v))
+                            else:
+                                for k, v in update['data'].items():
+                                    redis_v = pipe.hget(f"{self.key}-{update['feed']}-{update['symbol']}", k)
+                                    if redis_v is not None and redis_v != str(v):
+                                        pipe = pipe.hset(f"{self.key}-{update['feed']}-{update['symbol']}", k, str(v))
```

Comment:
As we discussed at call, If we make the key in advance, we can get the two parts togather.